api/src/opentrons/protocol_api/validation.py

```
@@ -70,7 +70,7 @@ def ensure_deck_slot(deck_slot: Union[int, str]) -> DeckSlotName:
         raise TypeError(f"Deck slot must be a string or integer, but got {deck_slot}")
 
     try:
-        return DeckSlotName(str(deck_slot))
+        return DeckSlotName.from_primitive(deck_slot)
```

I think we need to gate this behind `"apiLevel": "2.15"` somehow. The current release of PAPIv2.14 can't accept slot names like `A1`, so we want to raise a helpful error if someone tries it, instead of an unhelpful internal error.

This doesn't necessarily need to happen in this PR, but we do need to do it before release. Your call whether you want to do it here or ticket it as a next step.